capstone/src/main/webapp/scripts/members.js

```diff
@@ -0,0 +1,103 @@
+function loadMembers() {
+  fetch(`/all-group-members`).then(response => response.json()).then((allGroupMembers) => {
+    const allMembers = allGroupMembers;
+    const memberGrid = document.getElementsByClassName("member-grid-container")[0];
+    memberGrid.innerHTML = '';
+    for (let i = 0; i < allMembers.length; i++) {
+      memberGrid.appendChild(createMemberComponents(allMembers[i]));
+    }
+  }).then(() => {
+    addMemberProfileListener();
+  });
+}
+
+function createMemberComponents(memberInfo) {
+  const memberImgDiv = document.createElement('div');
+  memberImgDiv.id = memberInfo.userId;
+  memberImgDiv.title = memberInfo.firstName + " " + memberInfo.lastName;
+  // If user doesn't have image, replace with blank small yellow profile circle 
+  if (memberInfo.profilePic == null || memberInfo.profilePic == "") {
+    memberImgDiv.className = "member-grid-item member-img-blank small-member-img";
+  } else {
+    memberImgDiv.className = "member-grid-item member-img-div small-member-img";
+    const memberImg = document.createElement('img');
+    memberImg.className = "member-img small-member-img";
+    memberImg.src = "serve?blob-key=" + memberInfo.profilePic;
+    memberImgDiv.append(memberImg);
+  }
+  return memberImgDiv;
+}
+
+// When group member clicked, show detailed member profile modal
+function addMemberProfileListener() {
+  const elements = document.getElementsByClassName('member-grid-item');
+  for (let i = 0; i < elements.length; i++) {
+    elements[i].addEventListener("click", function() {
+      showMemberProfile(this.id);
+    });
+  }
+}
+
+// Show group member profile (profile pic, name, and badges)
+function showMemberProfile(userId){
+  fetch(`/group-member?id=${userId}`).then(response => response.json()).then((memResponse) => {
+    const modalContent = document.getElementsByClassName("modal-content")[0];
+    modalContent.innerHTML = "<span class='close'>&times;</span>";
+    addModalListeners();
+    modalContent.appendChild(createMemberModal(memResponse));
+  }).then(() => {
+    const modal = document.getElementById("memberProfile");
+    modal.style.display = "block";
```

Comment:
Same as the comment above, since the previous then is not returning a promise, you should just put this after the modalContent.append....
If you are using the then's to separate content, you can break it up into functions instead

Reply:
moved to after modalContent.append